p2p/server.go

```diff
@@ -178,6 +191,8 @@ func (serv *ExchangeServer[H]) handleRequestByHash(hash []byte) ([]H, error) {
 		attribute.Int64("height", int64(h.Height()))),
 	)
 	span.SetStatus(codes.Ok, "")
+
+	serv.metrics.getServed(ctx, time.Since(startTime))
```

Comment:
Does it compile? Looks like you pass 2 args instead of 3 here. Also check success cases in other handlers

Reply:
It vas variadic